controllers/iamrole_controller.go

```diff
@@ -190,11 +190,16 @@ func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Reques
 
 		// If IRSA is enabled, make sure the service account has the needed annotations
 		saConsistent := false
-		if saExists, saName := utils.ParseIRSAAnnotation(ctx, iamRole); saExists {
-			// Get the service account in kubernetes
-			// If it exists, check the annotations are correct
-			if saSpec := k8s.NewK8sManagerClient(r.Client).GetServiceAccount(ctx, iamRole.Namespace, saName); saSpec != nil {
-				saConsistent = validation.CompareRoleIRSA(ctx, saSpec, *config.Props)
+		if saExists, saNames := utils.ParseIRSAAnnotation(ctx, iamRole); saExists {
+			for i := 0; i < len(saNames); i++ {
+				// Get the service account in kubernetes
+				// If it exists, check the annotations are correct
+				if saSpec := k8s.NewK8sManagerClient(r.Client).GetServiceAccount(ctx, iamRole.Namespace, saNames[i]); saSpec != nil {
+					saConsistent = validation.CompareRoleIRSA(ctx, saSpec, *config.Props)
+					if saConsistent {
```

Comment:
This should be `if !saConsistent`, we want to keep `saConsistent` false in case SA is not consistent right?

Reply:
Thanks for pointing this out.  Yes, you are right. Fixed